azure/scope/machinepool.go

```diff
@@ -128,9 +128,31 @@ func (m *MachinePoolScope) ScaleSetSpec() azure.ScaleSetSpec {
 		SpotVMOptions:                m.AzureMachinePool.Spec.Template.SpotVMOptions,
 		FailureDomains:               m.MachinePool.Spec.FailureDomains,
 		TerminateNotificationTimeout: m.AzureMachinePool.Spec.Template.TerminateNotificationTimeout,
+		NetworkInterfaces:            m.AzureMachinePool.Spec.Template.NetworkInterfaces,
+		AllocatePublicIPPrefix:       m.AzureMachinePool.Spec.Template.AllocatePublicIPPrefix,
+		PublicIPPrefixID:             m.PublicIPPrefixID(),
 	}
 }
 
+// PublicIPPrefixSpecs returns the specs for public IP prefixes to be associated with the VMSS.
+func (m *MachinePoolScope) PublicIPPrefixSpecs() []azure.PublicIPPrefixSpec {
+	if m.AzureMachinePool.Spec.Template.AllocatePublicIPPrefix {
+		return []azure.PublicIPPrefixSpec{
+			{
+				Name:         azure.GenerateMachinePoolPublicIPPrefixName(m.Name()),
+				PrefixLength: 28,
```

Comment:
PrefixLength is setting the CIDR mask size of the PublicIPPrefix that Azure will allocate when creating the PublicIPPrefix resource. I can move it to a constant, though I don't think that will lend additional clarity. I chose 28 because it is the largest available, but actually now I am second-guessing this choice. I thought we needed this to get instance level, standard SKU public IPs but it looks like that is not the case per this doc. I am going to try to remove all the public ip prefix stuff to see if the instance level public IPs works without these static public ip prefixes.

Reply:
😞 
I tried just using the method from the docs but it looks like I need to update the compute API version dependencies. I'll pull this logic out of this PR for now. Will need a new PR that bumps the compute API to the new version, which unfortunately has tons of breaking changes.

`Specified api-version 2021-11-01 does not meet the minimum required api-version 2022-03-01 to set this property skuOnPublicIPAddressConfiguration`